### PR TITLE
Add token metadata support

### LINKED ISSUE #801: vault_token metadata argument does not exist
### Terraform Version
Terraform v0.12.26
+ provider.aws v2.65.0
+ provider.template v2.1.2
+ provider.vault v2.11.0

### Affected Resource(s)
vault_token

### Terraform Configuration Files
We have this in one terraform state to configure an "unseal" vault cluster
```hcl
resource "vault_policy" "unsealer" {
  name = "unsealer"
  policy = <<EOT
# Allow encryption for respective vault domain
path "${vault_mount.unsealer.path}/encrypt/{{identity.entity.metadata.domain}}" {
  capabilities = [ "update" ]
}

# Allow decryption for respective vault domain
path "${vault_mount.unsealer.path}/decrypt/{{identity.entity.metadata.domain}}" {
  capabilities = [ "update" ]
}
EOT
}

resource "vault_token_auth_backend_role" "unsealer" {
  role_name = "unsealer"
  allowed_policies = [vault_policy.unsealer.name]
  orphan = true
  renewable = true
}
```
...Which is referenced in this configuration
```hcl
resource "vault_token" "unsealer" {
  role_name = data.terraform_remote_state.unsealer.outputs.unsealer_role
  display_name = replace(local.fqdn, ".", "-")

  policies = [data.terraform_remote_state.unsealer.outputs.unsealer_policy]
  metadata = {
    domain = local.fqdn
  }
  renewable = true
}
```

### Expected Behavior
Metadata is an option when creating a token in the cli, and should be an argument for the terraform resource 

### Actual Behavior
```sh
Error: Unsupported argument

  on unseal_token.tf line 23, in resource "vault_token" "unsealer":
  23:   metadata = {

An argument named "metadata" is not expected here.
```

### PR DESCRIPTION
**Pull request by benashz**
Updates for #811 

Closes #801 

Output from acceptance testing:

```
$ time make testacc TESTARGS='-v -test.count 1 -test.run TestResourceToken*'

==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test -v -v -test.count 1 -test.run TestResourceToken* -timeout 30m ./...

ok      github.com/hashicorp/terraform-provider-vault/util      0.421s [no tests to run]
=== RUN   TestResourceToken_basic
--- PASS: TestResourceToken_basic (1.91s)
=== RUN   TestResourceToken_import
--- PASS: TestResourceToken_import (1.98s)
=== RUN   TestResourceToken_full
--- PASS: TestResourceToken_full (1.61s)
=== RUN   TestResourceToken_lookup
--- PASS: TestResourceToken_lookup (1.58s)
=== RUN   TestResourceToken_expire
--- PASS: TestResourceToken_expire (15.58s)
=== RUN   TestResourceToken_renew
--- PASS: TestResourceToken_renew (25.69s)
PASS
ok      github.com/hashicorp/terraform-provider-vault/vault     49.139s
make testacc TESTARGS='-v -test.count 1 -test.run TestResourceToken*'  42.70s user 20.36s system 112% cpu 56.228 total


...
```

### CODE REVIEW

**Review by vinay-gopalan — COMMENTED** [review]
Looks great! Had a question regarding data casting

File: vault/resource_token.go [hunk 1]
```diff
@@ -199,6 +208,14 @@ func tokenCreate(d *schema.ResourceData, meta interface{}) error {
 		createRequest.Renewable = &renewable
 	}
 
+	if v, ok := d.GetOk("metadata"); ok {
+		d := make(map[string]string)
+		for k, val := range v.(map[string]interface{}) {
```
[the remark this answers is not in the document]
Comment (benashz) [reply vault/resource_token.go:213]: I believe we are good, the schema validation should have taken care of invalid inputs.

[the remark this answers is not in the document]
Comment (benashz) [reply vault/resource_token.go:213]: `TestResourceToken_basic` seems to prove this out.

[the remark this answers is not in the document]
Comment (vinay-gopalan) [reply vault/resource_token.go:213]: Gotcha, thanks for confirming!

**Pull request merged**